sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -524,6 +527,13 @@ def evaluate(
     cv = check_cv(cv, enforce_start_with_window=True)
     scoring = _check_scores(scoring)
 
+    if callbacks is None:
+        callbacks = CallbackList()
```

Comment:
Not a review, just clarifying few doubts.

1. What does `CallbackList` do (or expect to do) when user passed no callbacks?
2. When you initialise with no callbacks, do we not need to pass `forecaster` or `scores` (as of now, ignoring my other comment about scores)? Hopefully those will be available always? What happens in this case?